subsys/usb/device_next/usbd_ch9.c

```diff
@@ -171,6 +172,8 @@ static int sreq_set_configuration(struct usbd_context *const uds_ctx)
 		uds_ctx->ch9_data.state = USBD_STATE_CONFIGURED;
 	}
 
+	usbd_msg_pub_simple(uds_ctx, USBD_MSG_CONFIGURATION, setup->wValue);
```

Comment:
The fact that notifications are asynchronous make this somewhat fragile, because the connection speed can change between the publish and handling. If the device disconnects though (which is mandatory for the speed to change) any stale configuration notifications are no longer relevant anyway.

I consider this as a example to where the notifications should be synchronous.